### PR TITLE
WFFHCOHORT-303

### PR DESCRIPTION
**Pull request by shelleyMurphy**
Adding the ability to specify user-defined code systems by passing in a map of key to url. I'm making the assumption that we're never mapping a url to another url, and that if a user specifies a literal URL then that will be used as it is given. 

If we want a different format than key=value I'm open to that.

### CODE REVIEW

File: cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java
Comment (csandersdev) [inline cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:389]: @tabishop has [a PR](https://github.com/Alvearie/quality-measure-and-cohort-service/pull/104) open right now for updating parameters names to use underscores instead of camel case. Not sure which of you will come down first, but we need to make sure that we line these two changes up to be consistent.

File: cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java
Comment (csandersdev) [inline cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java:110]: Can we remove this TODO now?

**Review by csandersdev — APPROVED** [review]
I added a couple very minor points. If Tim's PR comes down first, this will need a small change. If you come down first, Tim will need to change. I'm approving and leaving it up to you guys to work that out.

File: cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java
Comment (tabishop) [reply cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:389]: Correct, we are going with lower_snake_case for input parameters to line up with the rest-service-framework so this should be custom_code_system to align.

File: cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java
Comment (tabishop) [inline cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:399]: Can you add some documentation instead of the value="TODO" here?

**Review by tabishop — CHANGES_REQUESTED** [review]
Just a couple small changes requested

**Review by msargentibm — APPROVED** [review]
Code changes seem fine. Was also wondering if the handful of TODOs can be cleaned up or if  they were left around purposely.

File: cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java
Comment (shelleyMurphy) [reply cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java:110]: Definitely!

File: cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java
Comment (shelleyMurphy) [reply cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:399]: Yup, missed it.

File: cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java
Comment (shelleyMurphy) [reply cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:390]: Alright, I can either make that change now or wait for the change in Tim's PR, but there doesn't feel like a consensus in this comment thread.

Comment (tabishop) [reply cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java:390]: I think Corey and I are in agreement so I would like it if you made the change in your PR for this parameter to be custom_code_system. Once my PR is merged, the input parameters will all be lower_snake_case. Mine is just waiting for another review before it goes in.

Comment (shelleyMurphy) [reply cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java]: alright, pushed.

**Review by tabishop — APPROVED** [review]
Looks good, thanks!

File: cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java
Comment (dkwasny-ibm) [inline cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java:52]: Pedantic: Is this an active TODO, or leftover?

File: cohort-util/src/test/java/com/ibm/cohort/valueset/ValueSetUtilTest.java
Comment (dkwasny-ibm) [inline cohort-util/src/test/java/com/ibm/cohort/valueset/ValueSetUtilTest.java:124]: Pedantic: I could be wrong, but I think the `containsAll` checks are a tad overkill.  I don't think these will ever fail while the `entrySet()` equality check passes.  You also may be able to get away with just one `assertEquals()` that compares the maps.

**Review by dkwasny-ibm — APPROVED** [review]
👍 Just some light pedantry.

File: cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java
Comment (shelleyMurphy) [reply cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java:52]: leftover, killing

File: cohort-util/src/test/java/com/ibm/cohort/valueset/ValueSetUtilTest.java
Comment (shelleyMurphy) [reply cohort-util/src/test/java/com/ibm/cohort/valueset/ValueSetUtilTest.java:124]: I didn't realize I'd left all three in here, definitely leftover from playing with a different concept of testing at one point.

**Pull request merged**
